Fixing the code and test to work with hive-3.x

sivaponting opened the pull request:
<!--
Thank you for submitting a pull request!

Please verify that:
* [ ] Code is up-to-date with the `main` branch.
* [ ] You've successfully built and run the tests locally.
* [ ] There are new or updated unit tests validating the change.

Refer to CONTRIBUTING.md for more details.
  https://github.com/HotelsDotCom/waggle-dance/blob/main/CONTRIBUTING.md
-->

### :pencil: Description


### :link: Related Issues

patduin reviewed (changes requested):
Hi!
Please update the CHANGELOG and add a description in the PR that will help us with reviewing.

sivaponting commented:
> Hi! Please update the CHANGELOG and add a description in the PR that will help us with reviewing.

Updated the changelog

patduin commented on CHANGELOG.md:
```diff
@@ -1,3 +1,23 @@
+## [4.0.0] - 2023-07-28
```
FYI: We haven't made any releases with Hive3 mostly because internally we're not ready to move to hive3 just yet. 
We'll have to change either the artifact id or version to reflect hive3 otherwise we'll get clashes if make a release of the main branch. 
Are you ok to build your own version based on the branch for now?

sivaponting replied:
No need to merge with master. But can merge with hive-3.x version

sivaponting replied:
FYI. All the integration tests are working fine

patduin commented on CHANGELOG.md:
```diff
@@ -1,3 +1,23 @@
+* `jakarta` - `6.0.0`.
+* `apache-commons` - `3.12.0`.
+
+### Fixed
```
Any functional changes or is this mostly a merge of master into this branch?

sivaponting replied:
Most of the dependencies renamed it. Like javax.servlet-api moved to jakarta.servlet-api

zhangbutao commented:
Thanks for this PR! I think it is good to megre this change on the main branch, as most users use the main branch in production env.

sivaponting commented:
> Thanks for this PR! I think it is good to megre this change on the main branch, as most users use the main branch in production env.

This branch has taken all the recent commits from the main branch except the Glue config changes. Better we can maintain 2 branches (main for hive 2.x and another one for hive 3.x) for some time

sivaponting commented:
HI @patduin / @zhangbutao ,
Expecting ur feedback for my earlier comment. Can we maintain 2 different versions for some time?

zhangbutao commented:
> HI @patduin / @zhangbutao , Expecting ur feedback for my earlier comment. Can we maintain 2 different versions for some time?

I am also just a waggle-dance user. If i understand correctly, the hive-3.x waggel-dance version with this PR can also be compatible with hive2 thrift metastore; that is to say, users do not need to install two waggle-dance version to connect different hive thrift metastore.

In addition, it seems that the waggle-dance community has been focusing on maintaining the main branch, and it is more reasonable to merge this PR into main branch. Then users can use main branch to connect different hive thrift metastore.

I also would like to listen more from the core developers.

sivaponting commented:
> > HI @patduin / @zhangbutao , Expecting ur feedback for my earlier comment. Can we maintain 2 different versions for some time?
> 
> I am also just a waggle-dance user. If i understand correctly, the hive-3.x waggel-dance version with this PR can also be compatible with hive2 thrift metastore; that is to say, users do not need to install two waggle-dance version to connect different hive thrift metastore.
> 
> In addition, it seems that the waggle-dance community has been focusing on maintaining the main branch, and it is more reasonable to merge this PR into main branch. Then users can use main branch to connect different hive thrift metastore.
> 
> I also would like to listen more from the core developers.

May i know who has the rights to merge the PR?

zhangbutao commented:
> May i know who has the rights to merge the PR?

@sivaponting I think @patduin can.

patduin commented on pom.xml:
```diff
@@ -203,6 +459,15 @@
             <complianceLevel>${jdk.version}</complianceLevel>
             <showWeaveInfo>true</showWeaveInfo>
+            <Xlint>ignore</Xlint>
+            <!-- IMPORTANT -->
```
Yeah had some error? can you add what went wrong and why Important please?

sivaponting replied:
This comment is for to work Lombok with aspectj

patduin commented:
> > HI @patduin / @zhangbutao , Expecting ur feedback for my earlier comment. Can we maintain 2 different versions for some time?
> 
> I am also just a waggle-dance user. If i understand correctly, the hive-3.x waggel-dance version with this PR can also be compatible with hive2 thrift metastore; that is to say, users do not need to install two waggle-dance version to connect different hive thrift metastore.
> 
> In addition, it seems that the waggle-dance community has been focusing on maintaining the main branch, and it is more reasonable to merge this PR into main branch. Then users can use main branch to connect different hive thrift metastore.
> 
> Therefore, I prefer to maintain only one(main) branch if we can review and test this PR carefully.
> 
> I also would like to listen more from the core developers.

Hi @zhangbutao and @sivaponting 

Thanks, both for the discussion, let me try to shed some light.
Yes, I can merge (and some more people from our company can).

I suggest we merge this into the Hive-3.x branch. If we want a release of the hive-3.x release (we'll have to add some prefix to the version to prevent clashes with the main branch). Let me know if this is desirable, we can work on this after your PR is merged. Full disclosure we, WD maintainers, don't actively run nor test (besides tests that come with the code) the hive-3.x branch.

Why not merge into 'main' and have Hive3 and 2 compatibility in one go?
This would be ideal, many moons ago we found compatibility issues though and (being a bit selfish as a company we focussed on Hive-2.x as that's the HMS version we're running internally). We maintainers of WD would love to have both hive 2/3 compatibility, but it will require some time and effort to test extensively. Hence I suggest keeping the Hive-3.x branch for now.

Let me know if this sounds reasonable.

I had a skim through the PR and it generally looks good but I will try to have a bit more in-depth review later this week.

Thanks
  Patrick

jmnunezizu commented:
Hi @sivaponting and @zhangbutao. Thanks for the contribution.

As @patduin mentioned, we are still using Hive 2.x internally, so we would not recommend making the integration with Hive 3.x the main release.

About 2 years ago, we started analysing the work required to move from Hive 2.x to Hive 3.x. This was done across all the different tools and services. We found some services that worked fine, others that didn't work (and were patched), and others that didn't work (and were not patched). Waggle Dance is in the latter category.

Our summary says:

**Forward compatibility:**

- Hive 2 client → "Hive 2" WD → Hive 3 metastore: OK
- Hive 2 client → "Hive 2" WD → Hive 2 and 3 metastores: OK

**Backward compatibility:**

- Hive 3 client → "Hive 2" WD → Hive 3 metastore: read OK
- Hive 3 client → "Hive 2" WD → Hive 2 metastore: FAIL (no Hive 2 databases shown)
- Hive 3 client → "Hive 2" WD → Hive 2 and 3 metastores: FAIL (no Hive 2 databases shown)

sivaponting commented:
> > > HI @patduin / @zhangbutao , Expecting ur feedback for my earlier comment. Can we maintain 2 different versions for some time?
> > 
> > 
> > I am also just a waggle-dance user. If i understand correctly, the hive-3.x waggel-dance version with this PR can also be compatible with hive2 thrift metastore; that is to say, users do not need to install two waggle-dance version to connect different hive thrift metastore.
> > In addition, it seems that the waggle-dance community has been focusing on maintaining the main branch, and it is more reasonable to merge this PR into main branch. Then users can use main branch to connect different hive thrift metastore.
> > Therefore, I prefer to maintain only one(main) branch if we can review and test this PR carefully.
> > I also would like to listen more from the core developers.
> 
> Hi @zhangbutao and @sivaponting
> 
> Thanks, both for the discussion, let me try to shed some light. Yes, I can merge (and some more people from our company can).
> 
> I suggest we merge this into the Hive-3.x branch. If we want a release of the hive-3.x release (we'll have to add some prefix to the version to prevent clashes with the main branch). Let me know if this is desirable, we can work on this after your PR is merged. Full disclosure we, WD maintainers, don't actively run nor test (besides tests that come with the code) the hive-3.x branch.
> 
> Why not merge into 'main' and have Hive3 and 2 compatibility in one go? This would be ideal, many moons ago we found compatibility issues though and (being a bit selfish as a company we focussed on Hive-2.x as that's the HMS version we're running internally). We maintainers of WD would love to have both hive 2/3 compatibility, but it will require some time and effort to test extensively. Hence I suggest keeping the Hive-3.x branch for now.
> 
> Let me know if this sounds reasonable.
> 
> I had a skim through the PR and it generally looks good but I will try to have a bit more in-depth review later this week.
> 
> Thanks Patrick

Tht's why i'm suggesting to maintain 2 different branches. If this PR got merged into hive-3.x other developers also can contribute on top of tht.

zhangbutao commented:
@patduin @jmnunezizu Thanks for detailed explanation. I already got what your thought that we could not make sure the hive2&hive3 work correctly and stably at the same time at present.

I am ok to maintain the two branchs main(hive2)/hive-3.x and merge this PR into hive-3.x.  Let other contributors to continue to contribute the hive-3.x branch and make it stable.

jmnunezizu commented:
Thank you both.
I discussed this with @patduin in more detail and agreed that having a hive3 branch is the preferred way forward. 

We can then use that branch to test additional scenarios that weren't tested by us. See below.
- Hive2 client -> Hive3 WD -> Hive2 HMS

I'll wait until @patduin reviews the code in more detail this week, and then proceed.

Thanks once again.

sivaponting commented:
Tested in local with spark-shell & WD 3.x in local. Here is the output

**Spark-Shell**

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/1ef628f8-143c-4f78-8ccc-8e69702bc61b)


**WD**

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/768a8369-0983-4d62-bf04-1aaba45c705b)

sivaponting commented:
Here is the exception if u try to connect the spark-shell with 2.x version with WD 3.x

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/d8f1ce16-84d6-4a77-b254-e15c3d5b9d6d)

sivaponting commented:
> Hi @sivaponting and @zhangbutao. Thanks for the contribution.
> 
> As @patduin mentioned, we are still using Hive 2.x internally, so we would not recommend making the integration with Hive 3.x the main release.
> 
> About 2 years ago, we started analysing the work required to move from Hive 2.x to Hive 3.x. This was done across all the different tools and services. We found some services that worked fine, others that didn't work (and were patched), and others that didn't work (and were not patched). Waggle Dance is in the latter category.
> 
> Our summary says:
> 
> **Forward compatibility:**
> 
> * Hive 2 client → "Hive 2" WD → Hive 3 metastore: OK
> * Hive 2 client → "Hive 2" WD → Hive 2 and 3 metastores: OK
> 
> **Backward compatibility:**
> 
> * Hive 3 client → "Hive 2" WD → Hive 3 metastore: read OK
> * Hive 3 client → "Hive 2" WD → Hive 2 metastore: FAIL (no Hive 2 databases shown)
> * Hive 3 client → "Hive 2" WD → Hive 2 and 3 metastores: FAIL (no Hive 2 databases shown)

Hi @jmnunezizu / @patduin ,
I'm trying to test the below scenario
spark-shell(2.x) -> WD(2.x) -> Hive 3 metastore. Getting the below exception. Is it because of KERBEROS? Need ur suggestion

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/ae1efa37-2173-41e7-bd8d-0fe93e17ba2c)

patduin commented on pom.xml:
```diff
@@ -125,60 +91,350 @@
+      <dependency>
+        <groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+        <version>${lombok.version}</version>
```
this should be a provided dependency: https://projectlombok.org/setup/maven

sivaponting replied:
Ok will set the scope as provided

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java:
```diff
@@ -1483,16 +1684,253 @@ public CacheFileMetadataResult cache_file_metadata(CacheFileMetadataRequest req)
+
+  @Override
+  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
+  public LockResponse get_lock_materialization_rebuild(String dbName, String tableName, long txnId) throws TException {
```
Should we sent the request to the correct client based on the dbName?

sivaponting replied:
@Override
  @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
  public LockResponse get_lock_materialization_rebuild(String dbName, String tableName, long txnId) throws TException {
    DatabaseMapping databaseMapping = databaseMappingService.primaryDatabaseMapping();
    LockResponse result = databaseMapping.getClient().get_lock_materialization_rebuild(dbName, tableName, txnId);
    return result;
  }

It's a override method. It's expecting dbName. Any suggestions? pls

patduin replied:
Check other methods where we use s dbName we need to transform it:
```
 DatabaseMapping mapping = databaseMappingService.databaseMapping(dbname);
mapping.transformInboundDatabaseName(dbname)
```
and also sent the call to the correct client.

Please check this for all methods that are added. Sometimes the database name is inside a complex object.

sivaponting replied:
Hi @patduin ,
Based on ur suggestion, modified the code. Pls check it

sivaponting replied:
Transformed the database name present in the request & response of complex object too

patduin replied:
thanks looking good, left some comments, almost there!

patduin reviewed (changes requested):
left some comments but looking good, please have a look.

sivaponting commented:
> patduin

Pushed the changes. Pls check it

sivaponting commented:
Hi @patduin, if everything is ok. pls merge it. Thanks

abhimanyugupta07 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,23 @@
+## [4.0.0] - 2023-07-28
+### Updated Dependency
+* `hive` updated to `3.1.3.` (was `3.1.2`).
+* `hadoop` updated to `3.3.6.` (was `3.1.0`).
+* `spring` updated to `2.7.13.` (was `2.0.4-RELEASE`).
+* `guava` updated to `31.1-jre.` (was `23.0`).
+* `guice` updated to `4.13.2.` (was `4.13.1`).
+* `junit` updated to `5.1.0.` (was `4.0`).
+* `aspectj` updated to `1.9.7.` (was `1.8.9`).
+* `hcommon-hive-metastore` updated to `1.4.2.` (was `1.2.3`).
```
```suggestion
* `hive` updated to `3.1.3` (was `3.1.2`).
* `hadoop` updated to `3.3.6` (was `3.1.0`).
* `spring` updated to `2.7.13` (was `2.0.4-RELEASE`).
* `guava` updated to `31.1-jre` (was `23.0`).
* `guice` updated to `4.13.2` (was `4.13.1`).
* `junit` updated to `5.1.0` (was `4.0`).
* `aspectj` updated to `1.9.7` (was `1.8.9`).
* `hcommon-hive-metastore` updated to `1.4.2` (was `1.2.3`).
```

abhimanyugupta07 commented on pom.xml:
```diff
@@ -203,6 +460,15 @@
+            </excludes>
+            <forceAjcCompile>true</forceAjcCompile>
+            <sources/>
+            <!-- IMPORTSNT -->
```
```suggestion
            <!-- IMPORTANT -->
```

sivaponting commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -9,31 +9,53 @@ on:
   CodeQL-Build:
 
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
No need to mention the version.

sivaponting replied:
I think the workflow is not using main.yml

abhimanyugupta07 replied:
It is not using main.yml. It will be better to use the same version of Ubuntu for both workflows. I think fixing the version is better to avoid any surprises when the OS upgrades are available.

sivaponting replied:
Updated the ubuntu version

sivaponting commented:
Hi @patduin , 
It failed at this stage. Seems like authentication issue. Pls help to resolve it

<img width="2133" alt="image" src="https://github.com/ExpediaGroup/waggle-dance/assets/49893883/64a6d067-6bab-46f0-8b2f-c193a642e57a">

sivaponting commented:
Hi @patduin , 
updated the profile is waggle-dance-rpm module

sivaponting commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -511,6 +538,85 @@ public PartitionValuesRequest transformInboundPartitionValuesRequest(PartitionVa
+  @Override
+  public List<SQLUniqueConstraint> transformInboundSQLUniqueConstraints(List<SQLUniqueConstraint> sqlUniqueConstraints) {
+    for(SQLUniqueConstraint sqlUniqueConstraint: sqlUniqueConstraints) {
+      sqlUniqueConstraint.setTable_db(transformInboundDatabaseName(sqlUniqueConstraint.getTable_db()));
```
Hi @patduin , 
This snippet of code is changing the dbName. Any suggestions?

  public List<SQLUniqueConstraint> transformInboundSQLUniqueConstraints(List<SQLUniqueConstraint> sqlUniqueConstraints) {
    for(SQLUniqueConstraint sqlUniqueConstraint: sqlUniqueConstraints) {
      **sqlUniqueConstraint.setTable_db(transformInboundDatabaseName(sqlUniqueConstraint.getTable_db()));**

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  }
+
+  @Override
+  public ISchemaName transformOutboundISchemaName(ISchemaName iSchemaName) {
```
inline this method if only used in one place

sivaponting replied:
can we keep it in separate method so tht it will be clean?

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  @Override
+  public AddUniqueConstraintRequest transformInboundAddUniqueConstraintRequest(AddUniqueConstraintRequest request) {
+    for(SQLUniqueConstraint sqlUniqueConstraint: request.getUniqueConstraintCols()) {
+      sqlUniqueConstraint.setTable_db(transformOutboundDatabaseName(sqlUniqueConstraint.getTable_db()));
```
should be inbound

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  @Override
+  public AddNotNullConstraintRequest transformInboundAddNotNullConstraintRequest(AddNotNullConstraintRequest request) {
+    for(SQLNotNullConstraint sqlNotNullConstraint: request.getNotNullConstraintCols()) {
+      sqlNotNullConstraint.setTable_db(transformOutboundDatabaseName(sqlNotNullConstraint.getTable_db()));
```
inbound

sivaponting replied:
Resolved

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  @Override
+  public AddDefaultConstraintRequest transformInboundAddDefaultConstraintRequest(AddDefaultConstraintRequest request) {
+    for(SQLDefaultConstraint sqlDefaultConstraint: request.getDefaultConstraintCols()) {
+      sqlDefaultConstraint.setTable_db(transformOutboundDatabaseName(sqlDefaultConstraint.getTable_db()));
```
inbound

sivaponting replied:
Resolved

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  @Override
+  public AddCheckConstraintRequest transformInboundAddCheckConstraintRequest(AddCheckConstraintRequest request) {
+    for(SQLCheckConstraint sqlCheckConstraint: request.getCheckConstraintCols()) {
+      sqlCheckConstraint.setTable_db(transformOutboundDatabaseName(sqlCheckConstraint.getTable_db()));
```
inbound

sivaponting replied:
Resolved

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+
+  @Override
+  public SchemaVersionDescriptor transformInboundSchemaVersionDescriptor(SchemaVersionDescriptor request) {
+    request.getSchema().setDbName(transformInboundDatabaseName(request.getSchema().getDbName()));
```
can getSchema be null?

sivaponting replied:
Resolved

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+
+  @Override
+  public MapSchemaVersionToSerdeRequest transformInboundMapSchemaVersionToSerdeRequest(MapSchemaVersionToSerdeRequest request) {
+    request.getSchemaVersion().getSchema().setDbName(transformInboundDatabaseName(
```
can getSchemaVersion be null

sivaponting replied:
Resolved

sivaponting commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java:
```diff
@@ -525,6 +631,170 @@ public List<SchemaVersion> transformOutboundSchemaVersions(List<SchemaVersion> s
+  @Override
+  public AddForeignKeyRequest transformInboundAddForeignKeyRequest(AddForeignKeyRequest request) {
+    for(SQLForeignKey sqlForeignKey: request.getForeignKeyCols()) {
+      sqlForeignKey.setPktable_db(transformOutboundDatabaseName(sqlForeignKey.getPktable_db()));
```
Resolved

sivaponting commented:
Hi @patduin , 
Based on ur comments, modified the code and committed. Pls verify. Thanks

patduin commented:
> Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks

There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109

sivaponting commented:
> > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> 
> There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109

Can u re-trigger the flow. All tests were working fine for the latest commit

<img width="1421" alt="image" src="https://github.com/ExpediaGroup/waggle-dance/assets/49893883/682b3835-eef5-4f18-8ce9-f99635d36702">

sivaponting commented:
> > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> 
> There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109

Can u clear the cache and try

patduin commented:
> > > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> > 
> > 
> > There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109
> 
> Can u re-trigger the flow. All tests were working fine for the latest commit
> 
> <img alt="image" width="1421" src="https://user-images.githubusercontent.com/49893883/258366224-682b3835-eef5-4f18-8ce9-f99635d36702.png">

That's a code scan that's not running the actual tests.

sivaponting commented:
> > > > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> 
> > > 
> 
> > > 
> 
> > > There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109
> 
> > 
> 
> > Can u re-trigger the flow. All tests were working fine for the latest commit
> 
> > 
> 
> > <img alt="image" width="1421" src="https://user-images.githubusercontent.com/49893883/258366224-682b3835-eef5-4f18-8ce9-f99635d36702.png">
> 
> 
> 
> That's a code scan that's not running the actual tests.

I added the mvn verify step

sivaponting commented:
> > > > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> 
> > > 
> 
> > > 
> 
> > > There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109
> 
> > 
> 
> > Can u re-trigger the flow. All tests were working fine for the latest commit
> 
> > 
> 
> > <img alt="image" width="1421" src="https://user-images.githubusercontent.com/49893883/258366224-682b3835-eef5-4f18-8ce9-f99635d36702.png">
> 
> 
> 
> That's a code scan that's not running the actual tests.

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/b80367c0-666b-4f65-bb16-cab2eb3c7d77)

sivaponting commented:
> > > > Hi @patduin , Based on ur comments, modified the code and committed. Pls verify. Thanks
> 
> > > 
> 
> > > 
> 
> > > There was some test failure can you have a look and see if you can reproduce locally please? https://github.com/ExpediaGroup/waggle-dance/actions/runs/5751308549/job/15614306109
> 
> > 
> 
> > Can u re-trigger the flow. All tests were working fine for the latest commit
> 
> > 
> 
> > <img alt="image" width="1421" src="https://user-images.githubusercontent.com/49893883/258366224-682b3835-eef5-4f18-8ce9-f99635d36702.png">
> 
> 
> 
> That's a code scan that's not running the actual tests.

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/9ccf31bd-28f3-427d-92d1-6076ca469134)

sivaponting commented:
Hi @patduin ,
Could u pls clear the cache and try

![image](https://github.com/ExpediaGroup/waggle-dance/assets/49893883/ef44972f-1edb-45a2-a0f8-043ad6b1c57e)

sivaponting commented:
Hi @patduin ,

Here is the screenshot of the failure message. The issue is intermittent. Sometimes it occurs even in local. Any suggestions? 

<img width="1730" alt="image" src="https://github.com/ExpediaGroup/waggle-dance/assets/49893883/2c5df28e-dc1c-4fd5-87b2-51f0cb0ad4ce">

sivaponting commented:
Hi @patduin,
Intermittently these 2 methods are getting failed in Integration Test
1. getTableMeta()
2. typicalgetAllFunctions()

I suspect bcoz of these config configured as below. Shall we try increasing the timeout?

<img width="864" alt="image" src="https://github.com/ExpediaGroup/waggle-dance/assets/49893883/66a99fe6-b628-437a-9bbb-403f14a17979">

patduin commented:
> Hi @patduin, Intermittently these 2 methods are getting failed in Integration Test
> 
> 1. getTableMeta()
> 2. typicalgetAllFunctions()
> 
> I suspect bcoz of these config configured as below. Shall we try increasing the timeout?

Not sure but I verified they run ok independently. It's not a blocker.

patduin commented on waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/WaggleDanceException.java:
```diff
@@ -15,15 +15,10 @@
 
+import lombok.experimental.StandardException;
+
+@StandardException
```
Could you please revert these, I'm having issues in my environment compiling this, not sure why the other annotations are find and we're using lombok internally in other projects. But since this is an experimental features maybe it's best we don't rely on that.

sivaponting replied:
Reverted

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/NoPrimaryMetastoreException.java:
```diff
@@ -15,12 +15,11 @@
 
+import lombok.experimental.StandardException;
+
+@StandardException
```
please revert

sivaponting replied:
Reverted

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/WaggleDanceServerException.java:
```diff
@@ -15,20 +15,16 @@
  * request
  */
+
+@StandardException
```
please revert

sivaponting replied:
Reverted

patduin commented on waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/NotAllowedException.java:
```diff
@@ -15,16 +15,16 @@
  * Thrown when Waggle Dance encounters and operation that is not allowed by configured {@link AccessControlHandler}
  */
+
+@StandardException
```
please revert

sivaponting replied:
Reverted

patduin reviewed (changes requested):
this looks good just revert the @standardException lombok annotation please and I'll approve

sivaponting commented:
Hi @patduin ,
Reverted all @StandardException changes. pls help to approve. Thanks

patduin commented:
> Hi @patduin , Reverted all @StandardException changes. pls help to approve. Thanks

Thanks a lot for the contribution!

sivaponting commented:
> > Hi @patduin , Reverted all @StandardException changes. pls help to approve. Thanks
> 
> 
> 
> Thanks a lot for the contribution!

Welcome..
Thanks for the support @patduin/@abhimanyugupta07/ @jmnunezizu / @zhangbutao
